posthog/hogql_queries/query_runner.py

```diff
@@ -349,53 +373,89 @@ def _limit_context_aliased_for_cache(self) -> LimitContext:
     def calculate(self) -> R:
         raise NotImplementedError()
 
+    def enqueue_async_calculation(
+        self, *, cache_key: str, refresh_requested: bool = False, user: Optional[User] = None
+    ) -> QueryStatusResponse:
+        query_status = enqueue_process_query_task(
+            team=self.team,
+            user=user,
+            query_json=self.query.model_dump(),
+            query_id=self.query_id or cache_key,  # Use cache key as query ID to avoid duplicates
```

Comment:
I like this, deduplication for free.

Though side note: the term `query_id` is horribly overloaded, because I had to dig in just to make sure this is sets only the _async query status_ ID, and not the _ClickHouse query_ ID. I think we could disambiguate by renaming the former from "query" to "execution". E.g. `QueryStatusManager` →  `ExecutionStatusManager`, `query_id` →  `execution_id`. But best done in a separate PR